common/persistence/execution_manager.go

```diff
@@ -764,6 +765,12 @@ func (m *executionManagerImpl) GetCurrentExecution(
 	request *GetCurrentExecutionRequest,
 ) (*GetCurrentExecutionResponse, error) {
 	response, respErr := m.persistence.GetCurrentExecution(ctx, request)
+
+	var notFound *serviceerror.NotFound
+	if errors.As(respErr, &notFound) {
+		// strip persistence-specific error message
+		respErr = serviceerror.NewNotFound("workflow not found")
```

Comment:
```suggestion
		respErr = serviceerror.NewNotFound("current workflow execution not found")
```

Reply:
👍 right, that's more accurate!

Reply:
I don't think

> current workflow execution not found

works as a user-facing error message for clients. They're doing things like

```
workflowHandle.signal(...)
```

The error message is probably telling them that the workflow doesn't exist at all in the namespace they sent their request to, so "current" feels wrong. I'd vote for

```
workflow not found
```

Reply:
I was wondering if the right thing to do is to find an accurate message here (which is "current workflow execution not found") and make it more user-friendly one a case-by-case basis depending on the context ("workflow not found")

Reply:
I thought the more specific message would be better to differentiate `GetCurrentExection` from `GetWorkflowExecution` and other operations that might target a specific runID, since it might be possible the workflow exists but we failed to get the current execution record. But I don't feel that strongly about it, your logic also makes sense.

Reply:
Maybe return the more generic string and log a message with the specific API and workflow ID that failed?

Reply:
Let me see what `GetWorkflowExecution` is returning and we can make sure what we pick makes sense.